core-blocks/file/editable-link.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+
+export default class FileBlockEditableLink extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			showPlaceholder: ! this.props.text,
+		};
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.text !== this.props.text ) {
+			this.setState( { showPlaceholder: ! this.props.text } );
```

Comment:
What are the trade-offs of using state here vs. just using the `text` prop for rendering logic?

Reply:
Since I'm resorting to use a content-editable here, this is not a "controlled" component and therefore the `text` prop isn't updated on each keystroke.